service/history/timer_queue_standby_task_executor.go

```diff
@@ -321,6 +321,11 @@ func (t *timerQueueStandbyTaskExecutor) executeActivityRetryTimerTask(
 			return nil, nil
 		}
 
+		if activityInfo.Stamp != task.Stamp {
```

Comment:
standby activity retry timer may also push activity to matching, we also need to specify stamp there

Reply:
And the transfer task.

Reply:
done for pushActivity, for standby and transfer.

Reply:
keeping this open in case there is something else